katello/katello/hostname-change.rb

```diff
@@ -211,6 +233,22 @@ def run
 
       self.precheck
 
+      scenario_answers = load_scenario_answers(@options[:scenario])
+
+      unless custom_cert_options_check(scenario_answers)
+        fail_with_message("You are currently using custom certificates but not all custom certificate arguments are present. To change the hostname, " \
+                          "you need to provide certificates with the new hostname. Please review the custom certificate arguments required with --help.")
+      end
+
+      if @using_custom_certs && !@foreman_proxy_content
```

Comment:
The interaction here a bit hard to follow. The above check can fail if the options aren't provided but here you exclude if for proxies. Does that mean we should also exclude it above?